projects/cli-files/go-ls/cmd/root.go

```diff
@@ -0,0 +1,86 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+)
+
+func Execute() error {
+	pathArr := os.Args[1:]
+
+	var path string
+	var comma bool
+
+	if len(pathArr) == 0 {
+		path = "."
+		directoryOutput(path, comma)
+		return nil
+	}
+
+	if pathArr[0] == "-m" {
+		comma = true
+	} else {
+		comma = false
+	}
+
+	if len(pathArr) == 1 && comma {
+		path = "."
+		directoryOutput(path, comma)
+		return nil
+	}
+
+	var i int
+	if comma {
+		i = 1
+	} else {
+		i = 0
+	}
+
+	for i < len(pathArr) {
+
+		isPathValid, err := os.Stat(pathArr[i])
+		if err != nil {
+			return fmt.Errorf("%s no such file or directory", pathArr[i])
```

Comment:
Do we know the error was caused by the file/directory not existing? Or is it possible something else went wrong?

In general, unless we're confident we understand what went wrong (or we're actively trying to hide information, which is also sometimes valid), it's useful to preserve original errors in what we display rather than replacing them.